arch/armv7a/tls.S

```diff
@@ -0,0 +1,41 @@
+/*
+ * Phoenix-RTOS
+ *
+ * libphoenix
+ *
+ * tls access function
+ *
+ * Copyright 2023 Phoenix Systems
+ * Author: Hubert Badocha
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+
+ /* NOTE:
+ * When libphoenix is compiled as a PIC, even when linked into a static NOPIC binary,
+ * compiler generates accesses to TLS as calls to this function.
+ * This functions is simple version just for static binaries purposes,
+ * in dynamic binaries dynamic linker handles TLS accesses. */
+
+
+/* typedef struct {
+ *   unsigned long ti_moduleid;
+ *   unsigned long ti_tlsoffset;
+ * } TLS_index;
+ *
+ * void *__tls_get_addr(TLS_index *ti) */
+.globl __tls_get_addr;
+.type __tls_get_addr, %function;
```

Comment:
Ii it is to be in `thumb` mode then it is missing:
```suggestion
.thumb
.syntax unified

.text

.thumb_func
 .globl __tls_get_addr;
.type __tls_get_addr, %function;
.align 2
```
if `arm` mode:
```suggestion
.arm

.text

.globl __tls_get_addr;
.type __tls_get_addr, %function;
.align 2
```

Reply:
Thanks, it should be in thumb